nautobot/dcim/models/devices.py

```diff
@@ -535,6 +535,14 @@ class Device(PrimaryModel, ConfigContextModel, StatusModel):
     comments = models.TextField(blank=True)
     images = GenericRelation(to="extras.ImageAttachment")
 
+    secrets_group = models.ForeignKey(
+        to="extras.SecretsGroup",
+        on_delete=models.PROTECT,
```

Comment:
SET_NULL should be okay here - it would just mean that you can delete a SecretsGroup that has Devices and/or GitRepositories making use of it. I don't mind making this change if you'd prefer it that way.

Reply:
Personally I would do `SET_NULL` because the alternative would require the user to unlink every reference before they can delete, which is more annoying than it is helpful IMO.

Reply:
Sounds good, will do.